src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,166 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import duke.exception.TimeFormatException;
+import duke.exception.WrongUsageException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Class for task start with deadline
+ */
+public class Deadline extends Task {
+    private LocalDate by;
```

Comment:
This field name could be a bit ambiguous. Maybe a clearer alternative is `byDate` or `byTime`?